datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/application/ApplicationAuthorizationUtils.java

```diff
@@ -0,0 +1,26 @@
+package com.linkedin.datahub.graphql.resolvers.application;
+
+import static com.linkedin.metadata.Constants.APPLICATION_ENTITY_NAME;
+import static com.linkedin.metadata.authorization.ApiOperation.MANAGE;
+
+import com.datahub.authorization.AuthUtil;
+import com.linkedin.datahub.graphql.QueryContext;
+import java.util.List;
+import javax.annotation.Nonnull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ApplicationAuthorizationUtils {
+
+  private ApplicationAuthorizationUtils() {}
+
+  /**
+   * Returns true if the current user is authorized to edit any application entity. This is true if
+   * the user has either: 1. The EDIT_ENTITY privilege (super user privileges) 2. The
+   * EDIT_ENTITY_APPLICATIONS privilege
+   */
+  public static boolean canManageApplications(@Nonnull QueryContext context) {
+    return AuthUtil.isAuthorizedEntityType(
+        context.getOperationContext(), MANAGE, List.of(APPLICATION_ENTITY_NAME));
+  }
```

Comment:
so `EDIT_ENTITY_APPLICATIONS` is a metadata privilege on specific resources right? and same thing with `EDIT_ENTITY`. but this method is being used as a global thing like whether you can manage applications like CRUD on applications itself.

however i don't see any privileges in PoliciesConfig.java for application entities. I think I would also expect a global `MANAGE_APPLICATIONS_PRIVILEGE` that is a platform privilege instead here

Reply:
updated now- this piece wasn't pushed before. rather than introducing an application-specific EDIT_ENTITY_APPLICATIONS, we are just using EDIT_ENTITY. If a user has permission to EDIT_ENTITY on Applications as a whole, they can manage applications. Otherwise, they cannot manage applications.

Reply:
hmm gotcha. so now if someone should be able to manage applications they can have EDIT on the entity type in general, not specific entities of that type.